js/modules/k6/http/http.go

```diff
@@ -189,9 +196,18 @@ func (*HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.State,
 		Transport: state.HTTPTransport,
 		Timeout:   timeout,
 		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			max := int(state.Options.MaxRedirects.Int64)
-			if len(via) >= max {
-				return errors.Errorf("stopped after %d redirects", max)
+			max := state.Options.MaxRedirects.Int64
+			if redirects > -1 {
+				max = redirects
+			}
+			if len(via) > int(max) {
+				if redirects == -1 {
+					state.Logger.WithFields(log.Fields{
+						"error": fmt.Sprintf("stopped after %d redirects (controlled via option \"maxRedirects\")", max),
+						"url":   via[0].URL.String(),
+					}).Warn("Redirect Limit")
```

Comment:
Make this something like "Possible redirect loop, {{ via[0].Status }} response returned; pass `{ redirects: n }` to silence this". It doesn't matter if this is long, it should only be printed very rarely to make sure the user isn't accidentally doing the wrong thing.